File: EntangleMe/backend/app/schemas/quantum.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime

class QuantumTeleportRequest(BaseModel):
    sender_id: str = Field(..., description="ID of the sender")
    receiver_id: str = Field(..., description="ID of the receiver")
    classical_bit: int = Field(..., ge=0, le=1, description="Classical bit to teleport (0 or 1)")
    room_id: str = Field(..., description="Room ID where teleportation occurs")
    message_content: Optional[str] = Field(None, description="Optional text message")

class QuantumTeleportResponse(BaseModel):
    success: bool
    sender_id: str
    receiver_id: str
    sent_bit: int
    received_bit: int
    classical_bits: str = Field(..., description="Measurement results as binary string")
    receiver_state: str = Field(..., description="Final state of receiver qubit")
    teleportation_data: Dict[str, Any] = Field(..., description="Complete teleportation circuit data")
    timestamp: datetime
    message_id: Optional[str] = None

class QuantumCircuitData(BaseModel):
    circuit_diagram: str
    gate_sequence: list
    measurement_results: Dict[str, Any]
    final_state: str

class QuantumError(BaseModel):
    error: str
    details: Optional[str] = None

class TextTeleportRequest(BaseModel):
    receiver_id: str = Field(..., description="ID of the receiver")
    text_content: str = Field(..., description="Text message to teleport")
    room_id: Optional[str] = None
    group_id: Optional[str] = None
    is_direct: bool = False

File: EntangleMe/frontend/src/types/quantum.ts
export interface TeleportationResult {
  success: boolean;
  sent_bit: number;
  received_bit: number;
  classical_bits: string;
  receiver_state: string;
  teleportation_data: {
    circuit: any;
    measurements: any[];
    gates: any[];
    steps: any[];
  };
  success_probability: number;
  measurement_results: {
    classical_bits: string;
    received_bit: number;
    success: boolean;
  };
}

export interface CircuitVisualization {
  bit: number;
  circuit_data: {
    circuit_text: string;
    circuit_data: any;
    num_qubits: number;
    num_classical_bits: number;
    depth: number;
    gate_count: any;
  };
  description: string;
} 